### PR TITLE
Simplify code for validate_camera_efficiency

### PR DESCRIPTION
**Pull request by GernotMaier**
This PR updates the camera efficiency modules to take the recent changes in the handling of modeling and command line parameters into account:

- reading of parameters from the DB
- improved handling of command line parameters (remove duplication)
- moved code from the application into the camera_efficiency_module (this allows to have better testing and a clearer understanding of what inputs are needed)
- added unit tests (coverage for the camera efficiency modules was quite low)

Minor changes:

- fixed a couple of docstrings with obsolete entries (sorry, unrelated to this PR)

### CODE REVIEW

File: simtools/camera_efficiency.py [hunk 1]
```diff
@@ -24,120 +24,107 @@ class CameraEfficiency:
...
         """
-        if isinstance(tel, TelescopeModel):
-            self._logger.debug("TelescopeModel OK")
-            return tel
-
-        msg = "Invalid TelescopeModel"
-        self._logger.error(msg)
-        raise ValueError(msg)
+        zenith_angle = config_data.get("zenith_angle")
+        zenith_angle = zenith_angle.to("deg").value if zenith_angle is not None else 20.0
```
Comment (tobiaskleiner) [inline simtools/camera_efficiency.py:115]: Should we add a warning here if the values are set to default values?

File: tests/unit_tests/test_camera_efficiency.py [hunk 2]
```diff
@@ -50,26 +53,41 @@ def results_file(io_handler):
...
-    label = "test-from-kwargs"
-    zenith_angle = 30 * u.deg
-    ce = CameraEfficiency.from_kwargs(
-        telescope_model=tel_model,
-        site_model=site_model_north,
-        simtel_path=simtel_path,
-        label=label,
-        zenith_angle=zenith_angle,
-        test=True,
+def test_reprt(camera_efficiency_lst):
```
Comment (tobiaskleiner) [inline tests/unit_tests/test_camera_efficiency.py:56]: report?

**Review by tobiaskleiner — COMMENTED** [review]
Thanks @GernotMaier, looks good. See minor comments I have added.

File: tests/unit_tests/test_camera_efficiency.py (hunk 2 above)
Comment (GernotMaier) [reply tests/unit_tests/test_camera_efficiency.py:56]: Fixed.

**Review by tobiaskleiner — APPROVED** [review]

**Pull request merged**